test/unit/components/navigation.test.jsx

```diff
@@ -0,0 +1,69 @@
+const React = require('react');
+const {shallowWithIntl} = require('../../helpers/intl-helpers.jsx');
+import configureStore from 'redux-mock-store';
+const Navigation = require('../../../src/components/navigation/www/navigation.jsx');
+const sessionActions = require('../../../src/redux/session.js');
+
+describe('Navigation', () => {
+    const mockStore = configureStore();
+    let store;
+
+    beforeEach(() => {
+    });
+
+    const getNavigationWrapper = props => {
+        const wrapper = shallowWithIntl(
+            <Navigation
+                {...props}
+            />
+            , {context: {store}}
+        );
+        return wrapper
+            .dive() // unwrap redux connect(injectIntl(JoinFlow))
+            .dive(); // unwrap injectIntl(JoinFlow)
+    };
+
+    test('when using old join flow, clicking Join Scratch attemps to open registration', () => {
+        store = mockStore({
+            navigation: {
+                useScratch3Registration: false
+            },
+            session: {
```

Comment:
Do you need to set session and messageCount in these tests?

Reply:
i think we do need session (there's a conditional in render that checks it), but we don't need messageCount (which I think isn't even spelled right! Ack!)